packages/react-router/src/route.ts

```diff
@@ -71,10 +71,8 @@ export type RouteOptions<
   TRouterContext = AnyContext,
   TAllContext = AnyContext,
   TLoaderDeps extends Record<string, any> = {},
-  TLoaderDataReturn = unknown,
-  TLoaderData = [TLoaderDataReturn] extends [never]
-    ? undefined
```

Comment:
why can't we have `undefined` here anymore?
this is the return type in case the loader e.g. only throws a redirect